client/bench.go

```diff
@@ -0,0 +1,255 @@
+	multiplicandColumns := len(request.Multiplicand.GetColumns())
+
+	var v1 []float64
+	for i := 0; i < multiplierColumns; i++ {
+		for j := 0; j < multiplierRows; j++ {
+			v1 = append(v1, request.Multiplier.Columns[i].Coefficients[j])
+		}
+	}
+
+	var v2 []float64
+	for i := 0; i < multiplicandColumns; i++ {
+		for j := 0; j < multiplicandRows; j++ {
+			v2 = append(v2, request.Multiplicand.Columns[i].Coefficients[j])
+		}
+	}
+
+	pm.Multiplier = newMatrix(v1, multiplierColumns, multiplierRows)
+	pm.Multiplicand = newMatrix(v2, multiplicandColumns, multiplicandRows)
+}
+
+func (gn *goNumMessage) parseResponse(response matmult.MultiplyResponse) {
+	productRows := len(response.Result.GetColumns()[0].GetCoefficients())
+	productColumns := len(response.Result.GetColumns())
+
+	var v []float64
+	for i := 0; i < productColumns; i++ {
+		for j := 0; j < productRows; j++ {
+			v = append(v, response.Result.Columns[j].Coefficients[i])
+		}
+	}
+
+	gn.Product = *mat.NewDense(productRows, productColumns, v)
+}
+
+func (pm *primitiveMessage) parseResponse(response matmult.MultiplyResponse) {
+	productRows := len(response.Result.GetColumns()[0].GetCoefficients())
+	productColumns := len(response.Result.GetColumns())
+
+	var v []float64
+	for i := 0; i < productColumns; i++ {
+		for j := 0; j < productRows; j++ {
+			v = append(v, response.Result.Columns[j].Coefficients[i])
+		}
+	}
+
+	pm.Product = newMatrix(v, productColumns, productRows)
+}
+
+func multiplyOnSiteCPP(client matmult.MatrixProductAPIClient, message Writer) (*matmult.MultiplyResponse, error) {
+	request := message.writeRequest()
+	if len(request.GetMultiplier().GetColumns()) != len(request.GetMultiplicand().GetColumns()) {
+		err := errors.New("dimensions mismatch")
+		return nil, err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	return client.Multiply(ctx, &request)
+}
+
+func setup() (*grpc.ClientConn, matmult.MatrixProductAPIClient, matmult.MultiplyRequest, matmult.MultiplyResponse) {
+	conn, err := grpc.Dial(serverAddr, grpc.WithInsecure(), grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(math.MaxInt64)))
+	if err != nil {
+		log.Fatalf("Couldn't connect to the server: %v\n", err)
+	}
+	client := matmult.NewMatrixProductAPIClient(conn)
+
+	pbFile, err := os.Open(pbPath)
+	if err != nil {
+		log.Fatalf("Couldn't open protool buffers message: %v\n", err)
+	}
+	defer pbFile.Close()
+
+	pbBytes, err := ioutil.ReadAll(pbFile)
+	if err != nil {
+		log.Fatalf("Couldn't read protool buffers message: %v\n", err)
+	}
+
+	var pbRequest pbRequest
+	var request matmult.MultiplyRequest
+	if err := proto.Unmarshal(pbBytes, &request); err != nil {
+		log.Fatalf("Couldn't get request message: %v\n", err)
+	}
+	pbRequest.request = request
+
+	response, err := multiplyOnSiteCPP(client, pbRequest)
+	if err != nil {
+		log.Fatalf("Couldn't get a response: %v\n", err)
+	}
+
+	return conn, client, request, *response
```

Comment:
Why do you mutliply something here and then return request and response? What's the logic?